### DISCUSSION TITLE
Feature: Merchant/Buyer Login

### DISCUSSION THREAD

**Discussion by JuanIranzo** [uprm-inso4101-2024-2025-s1/semester-project-mercado-colegial-online#1] [Ideas]
I think one of the first features that should be discussed and implemented should be the login to the website. I think that there should be a separate login for sellers and buyers. If a person registers as a merchant, the system shall enable them to sign up/register as their brand and upload their inventory, set prices, order limits, delivery options, and product descriptions. If a person registers as a buyer then the system will take them directly to a home page where merchant inventory is featured. If this feature were to be implemented, then we should discuss a way for the system to determine or confirm that a merchant is in fact a student merchant with their brand.

**Comment by franrov4** [comment]
Authentification is key with our registration process i.e. asking the user for their student number. Having this restriccion will allow the user that chooses to be a seller, to be strictly from our campus, allowing a more standardized experience.

**Comment by david-baez-bravo** [comment]
It is very necessary to handle buyers and sellers separately. 

In the front end, to identify users vs buyers we should have them log in separately places. A place for sellers which would be like a dashboard where they can modify their inventory and prices and a place for buyers, would be like amazon, or any online shopping service. 

In the backend we need to have the sellers be able to communicate and make changes to the data server. The page for buyers would then reflect these changes. 

TLDR: Create separate dashboards for buyers and sellers. Handle them as different branches of the website.

**Reply by JuanIranzo** [reply]
This is exactly what I was thinking, partly because when thinking of how the DB tables will look like, seller tables could very possibly be different from buyer tables

**Comment by david-baez-bravo** [comment]
A way to sign up as a user and a way to sign up as a buyer needs to be implemented along with storing these credentials in the backend.

**Comment by elliot-gomez** [comment]
I completely agree with the importance of separate logins for buyers and merchants; I think this aspect of the project is the most important to get right. We could and should look to established platforms like Etsy, eBay, and Mercari for inspiration. These sites handle buyer and seller interactions and the switch between them well, and we can adapt their best practices to fit our needs. For authentication, we should require merchants to register with their UPR email or validate their student status through student IDs. Additionally, we should review other apps and projects developed here at UPRM and made for us students to see how they handle similar authentication processes, to ensure our approach is both secure and standard.
